articles/deploy-fleet-on-gcp.md

```diff
@@ -0,0 +1,586 @@
+# Deploy Fleet on GCP
+
+This Terraform project automates the deployment of Fleet on the Google Cloud Platform (GCP). It provisions a new GCP project (or uses an existing one if configured) and deploys all necessary infrastructure components, including:
+
+- A new Google Cloud Project (via terraform-google-modules/project-factory)
+- VPC Network and Subnets
+- Cloud NAT for outbound internet access
+- Cloud SQL for MySQL (database for Fleet)
+- Memorystore for Redis (caching for Fleet)
+- Google Cloud Storage (GCS) bucket (for Fleet software installers, S3-compatible access configured)
+- Google Secret Manager (for storing sensitive data like database passwords and Fleet server private key)
+- Cloud Run v2 Service (for the main Fleet application)
+- Cloud Run v2 Job (for database migrations)
+- External HTTP(S) Load Balancer with managed SSL certificate
+- Cloud DNS for managing the Fleet endpoint
+- Appropriate IAM service accounts and permissions
+
+## Prerequisites
+
+1. [Terraform](https://developer.hashicorp.com/terraform) version ~> 1.11.
+2. [Google Cloud SDK](https://cloud.google.com/sdk) installed and configured.
+3. GCP account & permissions:
+    - A Google Cloud organization ID.
+    - A billing account ID.
+    - Credentials configured for Terraform to interact with GCP. This typically means running   gcloud auth application-default login with a user account that has:
+    - Organization administrator or folder creator & project creator roles at the organization/  folder level (for the project-factory module).
+    - Billing account user role on the billing account.
+    - Once the project is created, subsequent applies for resources within the project will   generally require owner or editor on the provisioned project.
+    - Alternatively, use a service account with appropriate permissions.
+4. Registered domain name: You will need a domain name that you can manage DNS records for. This project will create a Cloud DNS Managed Zone for this domain (or a subdomain).
+5. (Optional) Fleet license key: If you have a Fleet Premium license, you can provide it.
+
+## Configuration
+
+1.  [Clone the repository](https://github.com/fleetdm/fleet-terraform/tree/e8d559ecba80ab35cb8014212f5bd0a22b545618/gcp) or ensure you have all the files. Alternatively, you can consume the project as a module. Below is a minimal example of how to declare the module in your Terraform configuration:
```

Comment:
@edwardsb, I took a punt at what you were asking for with your change request. Is this correct?